### PR TITLE
Implement Datagram & transaction handler for ICS04

### PR DESCRIPTION
**Pull request by junha1**
Depends on https://github.com/CodeChain-io/foundry/pull/218

### CODE REVIEW

File: core/src/ibc/transaction_handler/mod.rs [hunk 1]
```diff
@@ -132,6 +133,113 @@ pub fn execute(
...
+                            ibc::channel_04::types::ChannelOrder::UNORDERED
+                        }
+                    },
+                    raw.connection,
+                    raw.channel_identifier,
+                    raw.counterparty_channel_identifier,
+                    raw.version,
+                )
+                .map_err(|err| RuntimeError::IBC(format!("ChanOpenInit: {}", err)).into())
+                .map(|_| ())
```
Comment (majecty) [inline core/src/ibc/transaction_handler/mod.rs]: Why did you add ` .map(|_| ())` here?
How about this?

```
            channel_manager
                .chan_open_init(
                    {
                        if raw.order == 1 {
                            ibc::channel_04::types::ChannelOrder::ORDERED
                        } else {
                            ibc::channel_04::types::ChannelOrder::UNORDERED
                        }
                    },
                    raw.connection,
                    raw.channel_identifier,
                    raw.counterparty_channel_identifier,
                    raw.version,
                )
                .map_err(|err| RuntimeError::IBC(format!("ChanOpenInit: {}", err)).into())?;
            Ok(())
```

Comment (junha1) [reply core/src/ibc/transaction_handler/mod.rs]: That explicitly discards any returned result. I have no opinion about that, so I'll just follow your suggestion.

**Comment by majecty** [comment]
Good. Let's merge this PR after #218.
